File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.go
func findMin(nums []int) int {
	low, high := 0, len(nums)-1
	ans := nums[0]

	for low <= high {
		mid := low + (high-low)/2

		if nums[low] <= nums[high] {
			ans = min(ans, nums[low])
			break
		}
		if nums[low] <= nums[mid] {
			ans = min(ans, nums[low])
			low = mid + 1
		} else {
			ans = min(ans, nums[high])
			high = mid - 1
		}
	}

	return ans
}

func min(a int, b int) int {
	if a < b {
		return a
	}
	return b
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int s = 0, e = nums.size() - 1;
        int ans = nums[0];
        while (s <= e) {
            int m = e + (s - e) / 2;
            if(nums[s] <= nums[e]){
                ans = min(ans, nums[s]);
                break;
            }
            if (nums[s] <= nums[m]) {
                ans = min(ans, nums[s]);
                s = m + 1;
            } else {
                ans = min(ans, nums[m]);
                e = m - 1;
            }
        }
        return ans;
    }
};